Lib/logging/__init__.py

```diff
@@ -1543,9 +1556,15 @@ def isEnabledFor(self, level):
         """
         Is this logger enabled for level 'level'?
         """
-        if self.manager.disable >= level:
-            return False
-        return level >= self.getEffectiveLevel()
+        if level not in self._cache:
+            _acquireLock()
+            if self.manager.disable >= level:
+                self._cache[level] = False
+            else:
+                self._cache[level] = level >= self.getEffectiveLevel()
+            _releaseLock()
+
+        return self._cache[level]
```

Comment:
There is a risk of race condition if first you test if the key exists and then sets the key.

Why not using: "try: return self._cache[level] except KeyError: ..."?